### PR TITLE
Feat escape menu settings

### PR DESCRIPTION
**Pull request by apachano**
Authors:
@apachano 

## Summary of changes
- Extends settings menu to allow editing of all settings that get registered

## On approval
Merge

### CODE REVIEW

File: Phoenix/Common/Include/Common/Settings.hpp [hunk 1]
```diff
@@ -111,13 +111,15 @@ namespace phx
...
 		void setMax(int value);
+		int getMax() const {return m_maxValue;};
 
 		/**
 		 * @brief Set the minimum value for the setting.
 		 *
 		 * @param value The minimum value the setting can be.
 		 */
 		void setMin(int value);
+		int getMin() const {return m_minValue;};
```
Comment (beeperdeeper089) [inline Phoenix/Common/Include/Common/Settings.hpp:122]: Spacing, and you don't need the extra semicolon on the end.

File: Phoenix/Common/Include/Common/Settings.hpp [hunk 2]
```diff
@@ -126,6 +128,8 @@ namespace phx
 		 */
 		std::string getKey() const;
 
+		std::string getName() const;
```
Comment (beeperdeeper089) [inline Phoenix/Common/Include/Common/Settings.hpp:131]: Can we not return ``const std::string&``?

**Review by beeperdeeper089 — CHANGES_REQUESTED** [review]
I'm yet to test this myself but you deffo need to run a quick clang-format on this

File: Phoenix/Common/Include/Common/Settings.hpp (hunk 2 above)
Comment (apachano) [reply Phoenix/Common/Include/Common/Settings.hpp:131]: That would probably make sense

**Comment by apachano** [comment]
Fixed the consts and ran autoformat @beeperdeeper089

**Review by beeperdeeper089 — APPROVED** [review]

**Comment by beeperdeeper089** [comment]
lgtm, merge if you're done otherwise i'll re-review if u update

**Pull request merged**